<a href="https://colab.research.google.com/github/Loganathankumar/Tournament_Data_Analysis/blob/main/EDA_for_Tournament_Data_Preparation_and_Insights.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
import requests
import pandas as pd
import os

In [8]:
df = pd.read_excel(r"/content/sample_data/Input/Bangalore Cricket Tournament(1-137) (1).xlsx")

In [9]:
try:
    output_dir = r"/content/sample_data/Output/"
    output_path = os.path.join(output_dir, "Bangalore_Cricket_Tournament.csv")

    os.makedirs(output_dir, exist_ok=True)

    df.to_csv(output_path, index=False) # Changed 'df' to 'csv_team'
    print(f"Successfully saved to {output_path}")

except Exception as e:
    print(f"File Error: {e}")

Successfully saved to /content/sample_data/Output/Bangalore_Cricket_Tournament.csv


In [10]:
csv_team = pd.read_csv(r"/content/sample_data/Output/Bangalore_Cricket_Tournament.csv")
csv_team.head()

ID           Start time      Completion time                    Email  \
0   1  2025-11-18 11:59:10  2025-11-18 11:59:40  2000169906@hexaware.com   
1   2  2025-11-18 11:59:02  2025-11-18 11:59:47  2000126763@hexaware.com   
2   3  2025-11-18 11:59:05  2025-11-18 11:59:55  2000169331@hexaware.com   
3   4  2025-11-18 11:59:12  2025-11-18 11:59:56  2000147808@hexaware.com   
4   5  2025-11-18 11:59:04  2025-11-18 12:00:04  2000164927@hexaware.com   

                                     Name  Last modified time  \
0                     Balwant Singh Jaggi                 NaN   
1                       Gayatri Bongarala                 NaN   
2                       Lakshmi Priya C S                 NaN   
3  Chokkakula Veera Venkata Satyanarayana                 NaN   
4                            Umashankar N                 NaN   

                 Name2  Employee ID                              Email ID  \
0  Balwant Singh Jaggi   2000169906               2000169906@hexaware.com   
1    Gayatri Bongarala   2000126763                gayatrib2@hexaware.com   
2     Lakshmi Priya CS   2000169331               Lakshmip11@hexaware.com   
3         vvs narayana   2000147808  ChokkakulaVeeraVenkataS@hexaware.com   
4         N UMASHANKAR   2000164927             UmashankarN1@hexaware.com   

  GENDER  
0      M  
1      F  
2      F  
3      M  
4      M

In [11]:
print(f"Null values in 'Last modified time': {csv_team['Last modified time'].isnull().sum()}")

Null values in 'Last modified time': 137


In [12]:
# Drop the 'Last modified time' column
csv_team_drop_column = csv_team.drop(columns=['Last modified time', 'Name2'])
print(f"After dropping the unnecessary column")
csv_team_drop_column.head()

After dropping the unnecessary column


ID           Start time      Completion time                    Email  \
0   1  2025-11-18 11:59:10  2025-11-18 11:59:40  2000169906@hexaware.com   
1   2  2025-11-18 11:59:02  2025-11-18 11:59:47  2000126763@hexaware.com   
2   3  2025-11-18 11:59:05  2025-11-18 11:59:55  2000169331@hexaware.com   
3   4  2025-11-18 11:59:12  2025-11-18 11:59:56  2000147808@hexaware.com   
4   5  2025-11-18 11:59:04  2025-11-18 12:00:04  2000164927@hexaware.com   

                                     Name  Employee ID  \
0                     Balwant Singh Jaggi   2000169906   
1                       Gayatri Bongarala   2000126763   
2                       Lakshmi Priya C S   2000169331   
3  Chokkakula Veera Venkata Satyanarayana   2000147808   
4                            Umashankar N   2000164927   

                               Email ID GENDER  
0               2000169906@hexaware.com      M  
1                gayatrib2@hexaware.com      F  
2               Lakshmip11@hexaware.com      F  
3  ChokkakulaVeeraVenkataS@hexaware.com      M  
4             UmashankarN1@hexaware.com      M

# Task
Filter the 'csv_team_filtered' DataFrame to select the first 105 male entries and the first 15 female entries, sort each group by 'Start time' in ascending order, combine them into a new DataFrame called `scheduled_team`, and then display the head of `scheduled_team` along with a confirmation message.

**Reasoning**:
Filter the 'csv_team_drop_column' DataFrame to include only male and female entries, sort them by 'Start time', and select the first 105 rows as instructed in the subtask.



In [13]:
male_participants = csv_team_drop_column[csv_team_drop_column['GENDER'] == 'M'].sort_values(by='Start time', ascending=True).head(105)
print("First 105 male participants sorted by 'Start time':")
display(male_participants)

First 105 male participants sorted by 'Start time':


ID           Start time      Completion time                    Email  \
7      8  2025-11-18 11:58:51  2025-11-18 12:00:11  2000161559@hexaware.com   
4      5  2025-11-18 11:59:04  2025-11-18 12:00:04  2000164927@hexaware.com   
77    78  2025-11-18 11:59:08  2025-11-18 19:19:47  2000142513@hexaware.com   
0      1  2025-11-18 11:59:10  2025-11-18 11:59:40  2000169906@hexaware.com   
3      4  2025-11-18 11:59:12  2025-11-18 11:59:56  2000147808@hexaware.com   
..   ...                  ...                  ...                      ...   
106  107  2025-11-19 13:55:21  2025-11-19 13:56:11  1000072300@hexaware.com   
107  108  2025-11-19 13:59:17  2025-11-19 14:00:59  2000121174@hexaware.com   
109  110  2025-11-19 14:17:21  2025-11-19 14:18:20  2000130240@hexaware.com   
112  113  2025-11-19 15:55:42  2025-11-19 15:55:55  1000059523@hexaware.com   
114  115  2025-11-19 17:39:15  2025-11-19 17:39:48  2000156501@hexaware.com   

                                       Name  Employee ID  \
7                       Kondati Pavan Kumar   2000161559   
4                              Umashankar N   2000164927   
77                       Rajkumar Shanmugam   2000142513   
0                       Balwant Singh Jaggi   2000169906   
3    Chokkakula Veera Venkata Satyanarayana   2000147808   
..                                      ...          ...   
106                             Raja Ummadi   1000072300   
107                            Satish Kumar   2000121174   
109                                Mithun P   2000130240   
112                              Soorej T s   1000059523   
114                             Yogananda C   2000156501   

                                 Email ID GENDER  
7              KondatiPavanK@hexaware.com      M  
4               UmashankarN1@hexaware.com      M  
77                2000142513@hexaware.com      M  
0                 2000169906@hexaware.com      M  
3    ChokkakulaVeeraVenkataS@hexaware.com      M  
..                                    ...    ...  
106                    Rajau@hexaware.com      M  
107                 satishk7@hexaware.com      M  
109                 mithunp3@hexaware.com      M  
112                  soorejt@hexaware.com      M  
114               yoganandac@hexaware.com      M  

[105 rows x 8 columns]

In [14]:
female_participants = csv_team_drop_column[csv_team_drop_column['GENDER'] == 'F'].sort_values(by='Start time', ascending=True).head(15)
print("First 15 female participants sorted by 'Start time':")
display(female_participants)

First 15 female participants sorted by 'Start time':


ID           Start time      Completion time                    Email  \
1      2  2025-11-18 11:59:02  2025-11-18 11:59:47  2000126763@hexaware.com   
2      3  2025-11-18 11:59:05  2025-11-18 11:59:55  2000169331@hexaware.com   
39    40  2025-11-18 11:59:12  2025-11-18 12:24:06  2000171408@hexaware.com   
48    49  2025-11-18 12:00:30  2025-11-18 12:37:24  2000157731@hexaware.com   
47    48  2025-11-18 12:34:46  2025-11-18 12:35:19  2000097768@hexaware.com   
73    74  2025-11-18 16:58:04  2025-11-18 16:58:56  2000166698@hexaware.com   
88    89  2025-11-19 12:24:23  2025-11-19 12:24:52  2000080645@hexaware.com   
94    95  2025-11-19 12:29:11  2025-11-19 12:30:00  1000072687@hexaware.com   
108  109  2025-11-19 14:09:15  2025-11-19 14:10:38  2000171634@hexaware.com   
110  111  2025-11-19 14:49:00  2025-11-19 14:49:19  1000059526@hexaware.com   
113  114  2025-11-19 16:51:39  2025-11-19 16:54:33  2000161444@hexaware.com   
118  119  2025-11-19 22:17:16  2025-11-19 22:18:01  1000066680@hexaware.com   
122  123  2025-11-20 10:55:28  2025-11-20 10:56:34  2000147802@hexaware.com   
128  129  2025-11-20 14:08:09  2025-11-20 14:09:43  2000090347@hexaware.com   
136  137  2025-11-21 13:42:19  2025-11-21 13:43:00  2000159820@hexaware.com   

                   Name  Employee ID                  Email ID GENDER  
1     Gayatri Bongarala   2000126763    gayatrib2@hexaware.com      F  
2     Lakshmi Priya C S   2000169331   Lakshmip11@hexaware.com      F  
39   Sakhamudi Poojitha   2000171408   sakhamudip@hexaware.com      F  
48           Shobha Y L   2000157731   2000157731@hexaware.com      F  
47              Kavya C   2000097768      kavyac1@hexaware.com      F  
73           A. Sandiya   2000166698          A.S@hexaware.com      F  
88            Rachana N   2000080645   2000080645@hexaware.com      F  
94     Renuka Somagutta   1000072687     renukas3@hexaware.com      F  
108     Pathinti Ravali   2000171634    PathintiR@hexaware.com      F  
110   Shatabdi Khankary   1000059526    shatabdik@hexaware.com      F  
113      Supriya Kumari   2000161444   supriyak10@hexaware.com      F  
118        Neha Chauhan   1000066680       nehac1@hexaware.com      F  
122     Urvashi Virmani   2000147802     urvashiv@hexaware.com      F  
128      Sivakumar Nama   2000090347  SivakumarN1@hexaware.com      F  
136        Roopa Rani J   2000159820   RoopaRaniJ@hexaware.com      F

**Reasoning**:
Combine the `male_participants` and `female_participants` DataFrames into a new DataFrame called `scheduled_team`, and then display its head along with a confirmation message as per the subtask requirements.



In [15]:
scheduled_team = pd.concat([male_participants, female_participants])
print("Combined Male and Female Participants (scheduled_team):", len(scheduled_team))
scheduled_team.head()

Combined Male and Female Participants (scheduled_team): 120


ID           Start time      Completion time                    Email  \
7    8  2025-11-18 11:58:51  2025-11-18 12:00:11  2000161559@hexaware.com   
4    5  2025-11-18 11:59:04  2025-11-18 12:00:04  2000164927@hexaware.com   
77  78  2025-11-18 11:59:08  2025-11-18 19:19:47  2000142513@hexaware.com   
0    1  2025-11-18 11:59:10  2025-11-18 11:59:40  2000169906@hexaware.com   
3    4  2025-11-18 11:59:12  2025-11-18 11:59:56  2000147808@hexaware.com   

                                      Name  Employee ID  \
7                      Kondati Pavan Kumar   2000161559   
4                             Umashankar N   2000164927   
77                      Rajkumar Shanmugam   2000142513   
0                      Balwant Singh Jaggi   2000169906   
3   Chokkakula Veera Venkata Satyanarayana   2000147808   

                                Email ID GENDER  
7             KondatiPavanK@hexaware.com      M  
4              UmashankarN1@hexaware.com      M  
77               2000142513@hexaware.com      M  
0                2000169906@hexaware.com      M  
3   ChokkakulaVeeraVenkataS@hexaware.com      M

In [17]:
final_path = r"/content/sample_data/Output/blr_cricket_tournament_data_eda.json"

scheduled_team.to_json(os.path.join(final_path))
print(f"Successfully saved scheduled_team to {final_path}")

Successfully saved scheduled_team to /content/sample_data/Output/blr_cricket_tournament_data_eda.json


## Summary:

### Data Analysis Key Findings
*   Successfully created a `male_participants` DataFrame containing the first 105 male entries from `csv_team_drop_column`, sorted by 'Start time' in ascending order.
*   Successfully created a `female_participants` DataFrame containing the first 15 female entries from `csv_team_drop_column`, sorted by 'Start time' in ascending order.
*   The `scheduled_team` DataFrame was accurately constructed by concatenating the `male_participants` and `female_participants` DataFrames, resulting in a combined set of 120 entries (105 male and 15 female) each sorted by 'Start time' within their respective groups.

### Insights or Next Steps
*   The `scheduled_team` DataFrame is now prepared according to specific gender and quantity requirements, making it ready for scheduling or allocation purposes.
*   Further analysis could involve examining the 'Start time' distributions within the `scheduled_team` to identify any potential biases or clustering in the assigned times.


In [18]:
# To rename this Colab notebook file, click on the file name at the top-left corner of the interface and type in your desired name.